### PR TITLE
feature/transcription rule

### PR DESCRIPTION
**Pull request by sywhb**
- Add listenedAt to Page in elastic
- Update listenedAt in the API
- Add function to test if a page should be transcribed
- Add a pubsub subscription listening on the new article to automatically synthesize article

### CODE REVIEW

File: packages/api/src/routers/article_router.ts [hunk 1]
```diff
@@ -118,6 +119,13 @@ export function articleRouter() {
             audioUrl: existingSpeech.audioFileName,
             speechMarksUrl: existingSpeech.speechMarksFileName,
           })
+          await updatePage(
+            existingSpeech.elasticPageId,
+            {
+              listenedAt: new Date(),
```
Comment (jacksonh) [inline packages/api/src/routers/article_router.ts:125]: nice. This will probably be very useful

File: packages/api/src/services/speech.ts [hunk 2]
```diff
@@ -1,9 +1,77 @@
...
+export const shouldSynthesize = async (
+  userId: string,
+  page: Page
+): Promise<boolean> => {
+  const [recentListenedPage, count] = (await searchPages(
+    {
+      dateFilters: [
+        {
+          field: 'listenedAt',
+          startDate: new Date(Date.now() - 30 * 24 * 60 * 60 * 1000),
```
Comment (jacksonh) [inline packages/api/src/services/speech.ts:29]: 30 days, right?

**Review by jacksonh — APPROVED** [review]
Looks good. I wonder a bit about the elastic performance, but I guess its probably pretty good. Just a normal search.

File: packages/api/src/services/speech.ts (hunk 2 above)
Comment (sywhb) [reply packages/api/src/services/speech.ts:29]: Yes

**Pull request merged**